### PR TITLE
[Translation] throw again if non-excluded abstract services are tagged

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix https://github.com/symfony/symfony/pull/49850#issuecomment-1488742207
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Translation/DependencyInjection/TranslatorPass.php [hunk 1]
```diff
@@ -56,12 +56,10 @@ public function process(ContainerBuilder $container)
             $constraintVisitorDefinition = $container->getDefinition('translation.extractor.visitor.constraint');
             $constraintClassNames = [];
 
-            foreach ($container->getDefinitions() as $definition) {
-                if (!$definition->hasTag('validator.constraint_validator')) {
-                    continue;
-                }
+            foreach ($container->findTaggedServiceIds('validator.constraint_validator', true) as $definition) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Translation/DependencyInjection/TranslatorPass.php:59]: `as $id`

Comment (nicolas-grekas) [reply src/Symfony/Component/Translation/DependencyInjection/TranslatorPass.php:59]: ` as $id => $attributes` actually

**Comment by stof** [comment]
This is breaking the previous change, by not processing excluded services at all anymore, while they still needed to be processed for the discovery of available constraints.

This compiler was instentionally changed to process excluded service definitions (which are not returned anymore by `findTaggedServiceIds` in response to the comment you linked)

**Comment by nicolas-grekas** [comment]
@stof is correct.

**Pull request closed without merging**